Update WordPressAuthenticator in release 20.5

mokagio opened the pull request:
Following up on [this comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/19161#discussion_r940090779) in the 20.5 code freeze PR, this PR updates WordPressAuthenticator to the latest stable version, 2.1.1.

I explained why I didn't upgrade during code freeze in [that comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/19161#discussion_r940090779). While working on this PR, I tried to verified that 2.1.1 respects SemVer and integrates well with WordPress.

I was in particular curious to see whether the "What is WordPress.com" button, a new > 2.0.0 feature, would have appeared. It did not, which is good.

In hindsight, I could have controlled the diff between versions to learn more about the new code's behavior, see [this code in particular](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/compare/2.0.0...2.1.1#diff-210c95513e5b99c278b3f3a3e55f7a360908b842305ce90cdadb6a15fbf01ebdR32-R36). Still, I think this was a useful exercise, even though a time consuming one.


**WordPress 20.5.0.0, running WordPressAuthenticator 2.0.0**

<img alt="IMG_4681" src="https://user-images.githubusercontent.com/1218433/183787790-23646152-0c76-44b2-ba85-5a104afb546c.PNG" width=300/>

**WordPress `pr19163-524e05d`, running WordPressAuthenticator 2.0.1**

<img alt="IMG_37B0E6C0ADF3-1" src="https://user-images.githubusercontent.com/1218433/183788743-e53d73d3-f995-4ce7-b8ab-986d3bcaaa14.jpeg" width=300/>

**WordPress `pr19163-3740c92 `, running WordPressAuthenticator 2.1.1**

<img alt="IMG_EED300D50FC3-1" src="https://user-images.githubusercontent.com/1218433/183809686-9c647aa0-e850-4ed5-9d41-d6cb6cf3821f.jpeg" width=300/>

**WooCommerce 9.8.0.0, running WordPressAuthenticator 2.1.1** notice the "What is WordPress.com" link

<img alt="IMG_4680" src="https://user-images.githubusercontent.com/1218433/183787895-f3c7c9e4-1c61-454e-bf4d-103ac723df5f.PNG" width=300/>


## Regression Notes
1. Potential unintended areas of impact - The authentication screen and flow
2. What I did to test those areas of impact (or what existing automated tests I relied on) – I tried to login and succeeded
3. What automated tests I added (or what prevented me from doing so) – N.A.

PR submission checklist:

- [x] I have completed the Regression Notes. – N.A.
- [x] I have considered adding unit tests for my changes. – N.A.
- [x] I have considered adding accessibility improvements for my changes. – N.A.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. – N.A.

mokagio committed:
Update WordPressAuthenticator to version 2.0.1

mokagio committed:
Update WordPressAuthenticator to version 2.1.1

AliSoftware commented on Podfile:
```diff
@@ -222,7 +222,7 @@ abstract_target 'Apps' do
   pod 'Gridicons', '~> 1.1.0'
 
-  pod 'WordPressAuthenticator', '~> 2.0.0'
+  pod 'WordPressAuthenticator', '~> 2.1.0'
```
Should we make this `~> 2.1` instead of `~> 2.1.0`, so that SemVer would work (as the `~>` operator allows the last specified digit to increase but not the others, so `~> 2.1.0` would allow `2.1.1` but not `2.2.0`, while `~> 2.1` would.

AliSoftware replied:
Note that we should probably also consider doing the same (using `~> x.y` instead of `~> x.y.z`) for all other dependencies in the `Podfile` as well, though that larger/broader change for all pods would probably be for a separate PR.

AliSoftware reviewed (approved):
> I was in particular curious to see whether the "What is WordPress.com" button, a new > 2.0.0 feature, would have appeared. It did not, which is good.
>
> […] Still, I think this was a useful exercise, even though a time consuming one.

Great work and kudos on the deep investigation to check that this didn't break anything 👏 👍 💪 

I've left a suggestion to make the `~>` version specifier in the `Podfile` follow SemVer automatically (which would have made the `pod update WordPressAuthenticator` be bumped to `2.1.1` even if you kept `~> 2.0` in the `Podfile` to specify that 2.1 was not a hard requirement for the WP client app here). So I'll let you decide if you want to make that change before merging, but approving the PR nonetheless as this is not a blocker.

mokagio committed:
Use better SemVer requirement for WordPressAuthenticator pod

See discussion at
https://github.com/wordpress-mobile/WordPress-iOS/pull/19163#discussion_r942491647
